File: core/resource/VjoJavaLib/util/vjo/java/lang/Math.js
vjo.ctype('vjo.java.lang.Math') //< public final
.needs('vjo.java.lang.Double')
.needs('vjo.java.lang.MathUtil','')
.props({
    E:0, //< public final double
    PI:0, //< public final double
    //> public double abs(double d)
    abs:function(d){
        return Math.abs(d);
    },
    //> public double max(double a,double b)
    max:function(a,b){
        return Math.max(a,b);
    },
    //> public double min(double a,double b)
    min:function(a,b){
        return Math.min(a,b);
    },
    //> public double pow(double d,double d1)
    pow:function(d,d1){
        return Math.pow(d,d1);
    },
    //> public long round(double d)
    round:function(d){
        return Math.round(d);
    },
    //> public double sqrt(double d)
    sqrt:function(d){
        return Math.sqrt(d);
    },
    //> public double log10(double a)
    log10:function(a){
        return vjo.java.lang.MathUtil.log10(a);
    },
    //> public double cbrt(double a)
    cbrt:function(a){
        return vjo.java.lang.MathUtil.cbrt(a);
    },
    //> public double signum(double d)
    signum:function(d){
        return vjo.java.lang.MathUtil.signum(d);
    },
    //> public double sinh(double x)
    sinh:function(x){
        return vjo.java.lang.MathUtil.sinh(x);
    },
    //> public double cosh(double x)
    cosh:function(x){
        return vjo.java.lang.MathUtil.cosh(x);
    },
    //> public double tanh(double x)
    tanh:function(x){
        return vjo.java.lang.MathUtil.tanh(x);
    },
    //> public double hypot(double x,double y)
    hypot:function(x,y){
        return vjo.java.lang.MathUtil.hypot(x,y);
    },
    //> public double expm1(double x)
    expm1:function(x){
        return vjo.java.lang.MathUtil.expm1(x);
    },
    //> public double log1p(double x)
    log1p:function(x){
        return vjo.java.lang.MathUtil.log1p(x);
    },
    //> public double random()
    random:function(){
        return Math.random();
    },
    //> public double toRadians(double angdeg)
    toRadians:function(angdeg){
        return (angdeg/180)*this.PI;
    },
    //> public double toDegrees(double angrad)
    toDegrees:function(angrad){
        return (angrad*180)/this.PI;
    },
    //> public double sin(double x)
    sin:function(x){
        return Math.sin(x);
    },
    //> public double cos(double x)
    cos:function(x){
        return Math.cos(x);
    },
    //> public double tan(double x)
    tan:function(x){
        return Math.tan(x);
    },
    //> public double floor(double x)
    floor:function(x){
        return Math.floor(x);
    },
    //> public double ceil(double x)
    ceil:function(x){
        return Math.ceil(x);
    },
    //> public double exp(double x)
    exp:function(x){
        return Math.exp(x);
    },
    //> public double log(double x)
    log:function(x){
        return Math.log(x);
    },
    //> public double atan(double x)
    atan:function(x){
        return Math.atan(x);
    },
    //> public double asin(double x)
    asin:function(x){
        return Math.asin(x);
    },
    //> public double acos(double x)
    acos:function(x){
        return Math.acos(x);
    },
    //> public double atan2(double y,double x)
    atan2:function(y,x){
        return Math.atan2(y,x);
    },
    //> public double rint(double a)
    rint:function(a){
        return vjo.java.lang.MathUtil.rint(a);
    },
    //> public double IEEEremainder(double x,double p)
    IEEEremainder:function(x,p){
        var r=this.abs(x%p);
        if(this.vj$.Double.isNaN(r)||r===p || r<=parseFloat(this.abs(p)/2.0)){
            return r;
        }else {
            return this.signum(x)*(r-p);
        }
    }
})
.protos({
    //> private constructs()
    constructs:function(){
    }
})
.inits(function(){
    this.vj$.Math.E=Math.E;
    this.vj$.Math.PI=Math.PI;
})
.endType();

File: core/org.ebayopensource.vjet.core.jsnative/src/org/ebayopensource/dsf/jsnative/global/Math.java
/*******************************************************************************
 * Copyright (c) 2005-2011 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/
package org.ebayopensource.dsf.jsnative.global;

import org.ebayopensource.dsf.jsnative.anno.Function;
import org.ebayopensource.dsf.jsnative.anno.JsSupport;
import org.ebayopensource.dsf.jsnative.anno.JsVersion;
import org.ebayopensource.dsf.jsnative.anno.OverLoadFunc;
import org.ebayopensource.dsf.jsnative.anno.Property;
import org.ebayopensource.dsf.jsnative.anno.Static;

/**
 * 
 * Represents JavaScript native Math object
 *
 */
@JsSupport( {JsVersion.MOZILLA_ONE_DOT_ZERO, JsVersion.JSCRIPT_ONE_DOT_ZERO})
public interface Math extends Object {
	
	/**
	 * Euler's constant and the base of natural logarithms, approximately 2.718. 
	 */
	@Property @Static Number getE();
	
	/**
	 * The natural logarithm of 10, approximately 2.302.
	 */
	@Property @Static Number getLN10();
	
	/**
	 * The natural logarithm of 2, approximately 0.693.
	 */
	@Property @Static Number getLN2();
	
	/**
	 * The base 10 logarithm of E (approximately 0.434).
	 */
	@Property @Static Number getLOG10E();
	
	/**
	 * The base 2 logarithm of E (approximately 1.442).
	 */
	@Property @Static Number getLOG2E();
	
	/**
	 * The ratio of the circumference of a circle to its diameter, approximately 3.14159. 
	 */
	@Property @Static Number getPI();
	
	/**
	 * The square root of 1/2; equivalently, 1 over the square root of 2, approximately 0.707.  
	 */
	@Property @Static Number getSQRT1_2();
	
	/**
	 * The square root of 2, approximately 1.414. 
	 */
	@Property @Static Number getSQRT2();
	
	/**
	 * Returns the absolute value of a number.
	 */
	@OverLoadFunc @Static Number abs();
	
	/**
	 * Returns the absolute value of a number.
	 */
	@OverLoadFunc @Static Number abs(Number x);
	
	/**
	 * Returns the absolute value of a number.
	 */
	@OverLoadFunc @Static Number abs(String x);
	
	/**
	 * Returns the arccosine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number acos();
	
	/**
	 * Returns the arccosine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number acos(Number x);
	
	/**
	 * Returns the arccosine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number acos(String x);
	
	/**
	 * Returns the arcsine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number asin();
	
	/**
	 * Returns the arcsine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number asin(Number x);
	
	/**
	 * Returns the arcsine (in radians) of a number.
	 */
	@OverLoadFunc @Static Number asin(String x);
	
	/**
	 * Returns the arctangent (in radians) of a number.
	 */
	@OverLoadFunc @Static Number atan();
	
	/**
	 * Returns the arctangent (in radians) of a number.
	 */
	@OverLoadFunc @Static Number atan(Number x);
	
	/**
	 * Returns the arctangent (in radians) of a number.
	 */
	@OverLoadFunc @Static Number atan(String x);
	
	/**
	 * Returns the arctangent of the quotient of its arguments. 
	 */
	@Function @Static Number atan2(Object y, Object x);
	
	/**
	 * Returns the smallest integer greater than or equal to a number. 
	 */
	@OverLoadFunc @Static Number ceil();
	
	/**
	 * Returns the smallest integer greater than or equal to a number. 
	 */
	@OverLoadFunc @Static Number ceil(Number x);
	
	/**
	 * Returns the smallest integer greater than or equal to a number. 
	 */
	@OverLoadFunc @Static Number ceil(String x);
	
	/**
	 * Returns the cosine of a number.
	 */
	@OverLoadFunc @Static Number cos();
	
	/**
	 * Returns the cosine of a number.
	 */
	@OverLoadFunc @Static Number cos(Number x);
	
	/**
	 * Returns the cosine of a number.
	 */
	@OverLoadFunc @Static Number cos(String x);
	
	/**
	 * Returns Ex, where x is the argument, and E is Euler's constant, the base of the natural logarithms. 
	 */
	@OverLoadFunc @Static Number exp();
	
	/**
	 * Returns Ex, where x is the argument, and E is Euler's constant, the base of the natural logarithms. 
	 */
	@OverLoadFunc @Static Number exp(Number x);
	
	/**
	 * Returns Ex, where x is the argument, and E is Euler's constant, the base of the natural logarithms. 
	 */
	@OverLoadFunc @Static Number exp(String x);
	
	/**
	 * Returns the largest integer less than or equal to a number. 
	 */
	@OverLoadFunc @Static Number floor();
	
	/**
	 * Returns the largest integer less than or equal to a number. 
	 */
	@OverLoadFunc @Static Number floor(Number x);
	
	/**
	 * Returns the largest integer less than or equal to a number. 
	 */
	@OverLoadFunc @Static Number floor(String x);
	
	/**
	 * Returns the natural logarithm (base E) of a number. 
	 */
	@OverLoadFunc @Static Number log();
	
	/**
	 * Returns the natural logarithm (base E) of a number. 
	 */
	@OverLoadFunc @Static Number log(Number x);
	
	/**
	 * Returns the natural logarithm (base E) of a number. 
	 */
	@OverLoadFunc @Static Number log(String x);
	
	/**
	 * Returns the larger of two numbers.
	 */
	@OverLoadFunc @Static Number max();
	
	/**
	 * Returns the larger of two numbers.
	 */
	@OverLoadFunc @Static Number max(Object... values);
	
	/**
	 * Returns the larger of two numbers.
	 */
	@OverLoadFunc @Static Number min();
	
	/**
	 * Returns the larger of two numbers.
	 */
	@OverLoadFunc @Static Number min(Object... values);
	
	/**
	 * Returns base to the exponent power.
	 */
	@OverLoadFunc @Static Number pow();
	
	/**
	 * Returns base to the exponent power.
	 */
	@OverLoadFunc @Static Number pow(Number base, Number exponent);
	
	/**
	 * Returns base to the exponent power.
	 */
	@OverLoadFunc @Static Number pow(Number base, String exponent);
	
	/**
	 * Returns base to the exponent power.
	 */
	@OverLoadFunc @Static Number pow(String base, Number exponent);
	
	/**
	 * Returns base to the exponent power.
	 */
	@OverLoadFunc @Static Number pow(String base, String exponent);
	
	/**
	 * Returns a pseudo-random number in the range [0,1) -- that is, between 0 (inclusive) and 1 (exclusive). 
	 * The random number generator is seeded from the current time, as in Java. 
	 */
	@Function @Static Number random();
	
	/**
	 * Returns the value of a number rounded to the nearest integer. 
	 */
	@OverLoadFunc @Static Number round();
	
	/**
	 * Returns the value of a number rounded to the nearest integer. 
	 */
	@OverLoadFunc @Static Number round(Number x);
	
	/**
	 * Returns the value of a number rounded to the nearest integer. 
	 */
	@OverLoadFunc @Static Number round(String x);
	
	/**
	 * Returns the sine of a number.
	 */
	@OverLoadFunc @Static Number sin();
	
	/**
	 * Returns the sine of a number.
	 */
	@OverLoadFunc @Static Number sin(Number x);
	
	/**
	 * Returns the sine of a number.
	 */
	@OverLoadFunc @Static Number sin(String x);
	
	/**
	 * Returns the square root of a number.
	 */
	@OverLoadFunc @Static Number sqrt();
	
	/**
	 * Returns the square root of a number.
	 */
	@OverLoadFunc @Static Number sqrt(Number x);
	
	/**
	 * Returns the square root of a number.
	 */
	@OverLoadFunc @Static Number sqrt(String x);
	
	/**
	 * Returns the tangent of a number.
	 */
	@OverLoadFunc @Static Number tan();
	
	/**
	 * Returns the tangent of a number.
	 */
	@OverLoadFunc @Static Number tan(Number x);
	
	/**
	 * Returns the tangent of a number.
	 */
	@OverLoadFunc @Static Number tan(String x);
}
